src/mrinufft/trajectories/display.py

```diff
@@ -19,186 +26,588 @@
 NB_COLORS = len(COLOR_CYCLE)
 
 
-LINEWIDTH = 2
-POINTSIZE = 10
-FONTSIZE = 18
+class DisplayConfig:
+    """
+    A singleton used to share parameters related to display.
+
+    Attributes
+    ----------
+    alpha : float
+        Transparency used for area plots, by default 0.2.
+    linewidth : float
+        Width for lines or curves, by default 2.
+    pointsize : float
+        Size for points used to show constraints, by default 10.
+    fontsize : float
+        Font size for most labels and texts, by default 18.
+    small_fontsize : float
+        Font size for smaller texts, by default 14.
+    """
+
+    _instance = None
+
+    def __new__(cls):
+        """Rerouting new instances to the singleton."""
+        if cls._instance is None:
+            cls._instance = super().__new__(cls)
+            cls._instance.initialize_config()
+        return cls._instance
+
+    def initialize_config(self):
+        """Initialize the display configuration with default values."""
+        self.alpha = 0.2
+        self.linewidth = 2
+        self.pointsize = 10
+        self.fontsize = 18
+        self.small_fontsize = 14
+
+
+# Initialize DisplayConfig singleton for user convenience
+displayConfig = DisplayConfig()
+
+
+##############
+# TICK UTILS #
+##############
 
 
-def setup_2D_ticks(size, ax=None):
+def _setup_2D_ticks(figsize, fig=None):
     """Add ticks to 2D plot."""
-    if ax is None:
-        plt.figure(figsize=(size, size))
-        ax = plt.gca()
+    displayConfig = DisplayConfig()
+    if fig is None:
+        fig = plt.figure(figsize=(figsize, figsize))
+    ax = fig.subplots()
     ax.grid(True)
     ax.set_xticks([-KMAX, -KMAX / 2, 0, KMAX / 2, KMAX])
     ax.set_yticks([-KMAX, -KMAX / 2, 0, KMAX / 2, KMAX])
     ax.set_xlim((-KMAX, KMAX))
     ax.set_ylim((-KMAX, KMAX))
-    ax.set_xlabel("kx", fontsize=FONTSIZE)
-    ax.set_ylabel("ky", fontsize=FONTSIZE)
+    ax.set_xlabel("kx", fontsize=displayConfig.fontsize)
+    ax.set_ylabel("ky", fontsize=displayConfig.fontsize)
     return ax
 
 
-def setup_3D_ticks(size, ax=None):
+def _setup_3D_ticks(figsize, fig=None):
     """Add ticks to 3D plot."""
-    if ax is None:
-        fig = plt.figure(figsize=(size, size))
-        ax = fig.add_subplot(projection="3d")
+    displayConfig = DisplayConfig()
```

Comment:
No need for it, its a global of the module
```suggestion
```

Reply:
Done